messaging/src/main/java/org/axonframework/common/caching/Cache.java

```diff
@@ -59,6 +60,17 @@ public interface Cache {
      */
     boolean putIfAbsent(Object key, Object value);
 
+    /**
+     * Stores the value given by the {@code valueSupplier} in the cache, under given {@code key}, if no element is yet
+     * available under that key. This operation is performed atomically.
+     *
+     * @param key           The key under which to store the item
+     * @param valueSupplier A supplier that lazily supplies the value when necessary
+     * @return The value that is in the cache after the operation. This can be the original value or the one supplied by
+     * the {@code valueSupplier}.
+     */
+    <T> T getOrCompute(Object key, Supplier<T> valueSupplier);
```

Comment:
I think we should add a default using the `get` and `putIfAbsent` not a great default, but needed to be non breaking.